packages/components/virtual-list/src/hooks/useGridWheel.ts

```diff
@@ -0,0 +1,61 @@
+import { rAF, cAF } from '@element-plus/utils/raf'
+import { isFF } from '../utils'
+
+import type { ComputedRef } from 'vue'
+
+interface IWheelState {
+  atXStartEdge: ComputedRef<boolean>
+  atXEndEdge: ComputedRef<boolean>
+  atYStartEdge: ComputedRef<boolean>
+  atYEndEdge: ComputedRef<boolean>
+}
+
+type IWheelHandler = (x: number, y: number) => void
+
+export const useGridWheel = (
+  { atXEndEdge, atXStartEdge, atYEndEdge, atYStartEdge }: IWheelState,
+  onWheelDelta: IWheelHandler
+) => {
+  let frameHandle: number | null = null
```

Comment:
Maybe we should use `undefined`

Reply:
By removing `| null` I am able to remove `!` assertion for the variable.